test/integration/testpaths/testpath_volume_cuncurrent_snapshots.py

```diff
@@ -0,0 +1,827 @@
+            self.assertTrue(snaps.id in any(
+                snp.id) for snp in snapshots)
+
+        for snapshot in self.snapshot_pool:
+            self.assertTrue(
+                is_snapshot_on_nfs(
+                    self.apiclient,
+                    self.dbclient,
+                    self.config,
+                    self.zone.id,
+                    snapshot.id))
+
+        for snap in self.snapshot_pool:
+            snap.delete(self.apiclient)
+
+        self.snapshot_pool = []
+        # Step 2
+        thread_pool = []
+        for i in range(4):
+            try:
+                t = Thread(
+                    target=CreateSnapshot,
+                    args=(
+                        self,
+                        self.root_pool[i],
+                        False))
+                thread_pool.append(t)
+
+            except Exception as e:
+                raise Exception(
+                    "Warning: Exception unable to start thread : %s" %
+                    e)
+
+        for thread in thread_pool:
+            thread.start()
+
+        for thread in thread_pool:
+            thread.join()
+
+        snapshots = list_snapshots(
+            self.apiclient,
+            account=self.account.name,
+            domainid=self.account.domainid,
+            listall=True
+        )
+
+        for snaps in self.snapshot_pool:
+            self.assertTrue(snaps.id in any(
+                snp.id) for snp in snapshots)
+
+        for snapshot in self.snapshot_pool:
+            self.assertTrue(
+                is_snapshot_on_nfs(
+                    self.apiclient,
+                    self.dbclient,
+                    self.config,
+                    self.zone.id,
+                    snapshot.id))
+
+        for snap in self.snapshot_pool:
+            snap.delete(self.apiclient)
+
+        self.snapshot_pool = []
+
+        # Step 3
+        # Recurring snapshot
+        try:
+            t1 = Thread(
+                target=CreateSnapshot,
+                args=(
+                    self,
+                    self.root_pool[0],
+                    True))
+            t2 = Thread(
+                target=CreateSnapshot,
+                args=(
+                    self,
+                    self.root_pool[1],
+                    True))
+            t1.start()
+            t2.start()
+            t1.join()
+            t2.join()
+
+        except:
+            self.debug("Error: unable to start thread")
+
+        for rec_snap in self.rec_policy_pool:
+            list_snapshots_policy_1 = list_snapshot_policy(
+                self.apiclient,
+                id=rec_snap.id,
+            )
+
+            list_validation = validateList(list_snapshots_policy_1)
+
+            self.assertEqual(
+                list_validation[0],
+                PASS,
+                "snapshot list validation failed due to %s" %
+                list_validation[2])
+
+        time.sleep(3600)
```

Comment:
Don't use hard coded time. Make use of time which is used while creating recurring snapshot policy.